src/doc.cpp

```diff
@@ -727,11 +729,80 @@ void Doc::PrepareDrawing()
+            Syl *syl = new Syl();
+            Text *text = new Text();
+            syl->AddChild(text);
+            (*it)->AddChild(syl);
+        }
+    }
 
-    PrepareFacsimileParams prepareFacsimileParams(this->GetFacsimile());
-    Functor prepareFacsimile(&Object::PrepareFacsimile);
-    this->Process(&prepareFacsimile, &prepareFacsimileParams);
+    /************ Resolve @facs ************/
+    if (this->GetType() == Facs) {
+        // Associate zones with elements
+        PrepareFacsimileParams prepareFacsimileParams(this->GetFacsimile());
+        Functor prepareFacsimile(&Object::PrepareFacsimile);
+        this->Process(&prepareFacsimile, &prepareFacsimileParams);
+
+        // Add default syl zone if one is not present.
+        ListOfObjects syls;
+        ClassIdComparison comp(SYL);
+        this->FindAllDescendantByComparison(&syls, &comp);
+        for (auto it = syls.begin(); it != syls.end(); ++it) {
+            FacsimileInterface *fi = (*it)->GetFacsimileInterface();
+            assert(fi);
+            if (!fi->HasFacs()) {
+                Zone *zone = new Zone();
+                const int offsetUly = 100;
+                const int offsetLrx = 100;
+                const int offsetLry = 200;
+                LayerElement *syllable = dynamic_cast<LayerElement *>((*it)->GetFirstAncestor(SYLLABLE));
+                assert(syllable);
+                if (syllable->HasFacs()) {
+                    Zone *tempZone = syllable->GetZone();
+                    assert(tempZone);
+                    zone->SetUlx(tempZone->GetUlx());
+                    zone->SetUly(tempZone->GetUly() + offsetUly);
+                    zone->SetLrx(tempZone->GetLrx() + offsetLrx);
+                    zone->SetLry(tempZone->GetLry() + offsetLry);
+                }
+                else {
+                    int ulx, uly, lrx, lry;
+                    if (syllable->GenerateZoneBounds(&ulx, &uly, &lrx, &lry)) {
+                        if (ulx == 0 || uly == 0 || lrx == 0 || lry == 0) {
+                            LogWarning("Zero value when generating bbox from %s: (%d, %d, %d, %d)",
+                                syllable->GetUuid().c_str(), ulx, uly, lrx, lry);
+                        }
+                        zone->SetUlx(ulx);
+                        zone->SetUly(uly + offsetUly);
+                        zone->SetLrx(lrx + offsetLrx);
+                        zone->SetLry(lry + offsetLry);
+                    }
+                    else {
+                        LogWarning("Failed to create zone for %s of type %s", (*it)->GetUuid().c_str(),
+                            (*it)->GetClassName().c_str());
+                        delete zone;
+                        continue;
+                    }
+                }
+                Object *surface = this->GetFacsimile()->FindDescendantByType(SURFACE);
+                assert(surface);
+                surface->AddChild(zone);
+                fi->SetZone(zone);
+            }
```

Comment:
I tried adding it to `PrepareFacsimile` but since the zone is created using zone information from other elements in the syllable in many cases, those zones will need to be associated first. Is there a way to have a functor handle sibling elements first before returning to a specific element?

Reply:
Not quite. What you could add a PreperFacsimileEnd end functor and trigger the loop over m_zonelessSyls at the end. E.g., the end of the Doc, or the end of a System. But you can leave it like it is. This is just for the record.

Reply:
Ah okay thanks for letting me know!